src/app/services/transmart-services/transmart-resource.service.ts

```diff
@@ -255,24 +271,24 @@ export class TransmartResourceService {
           }, err => {
             reject(err);
           });
-      } else {
-        this.exclusionCounts.patientCount = 0;
-        this.exclusionCounts.observationCount = 0;
-        resolve(true);
       }
     });
   }
 
   updateInclusionCounts(constraint: Constraint, totalCountItem?: TransmartCountItem): Promise<any> {
     return new Promise<any>((resolve, reject) => {
       if (this.autosaveSubjectSets) {
-        if (totalCountItem) {
-          this.inclusionCounts.patientCount = totalCountItem.patientCount + this.exclusionCounts.patientCount;
-          this.inclusionCounts.observationCount = totalCountItem.observationCount + this.exclusionCounts.observationCount;
-          resolve(true);
+        if (!totalCountItem) {
+          return reject('Missing transmart total-count item to calculate inclusion counts.');
+        }
+        this.inclusionCounts.patientCount = totalCountItem.patientCount - this.exclusionCounts.patientCount;
```

Comment:
hmm, are you sure about the subtraction character? Previously `total = inclusion - exclusion`, so `inclusion = total + exclusion` (there was a `+` here).

Reply:
You are correct, I will fix it.